python/ray/air/checkpoint.py

```diff
@@ -40,6 +43,21 @@
 logger = logging.getLogger(__name__)
 
 
+@DeveloperAPI
+@dataclass
+class CheckpointMetadata:
+    checkpoint_type: Type["Checkpoint"]
+    checkpoint_state: Dict[str, Any]
+    ray_version: str = ray.__version__
```

Comment:
Should we store commit (`ray.__commit__`) instead of version? Version might not be sufficient for nightly.

Reply:
We could do that. Downside is that it might be harder to do checks like

```
if minor_version(metadata.ray_version) != minor_version(__version__):
    logging.warn("This checkpoint was created with another Ray version.")
```

We could also store both.